src/main/java/org/mule/extension/db/internal/result/statement/StatementResultIterator.java

```diff
@@ -187,15 +190,60 @@ protected SingleStatementResult processOutputParam() {
   protected SingleStatementResult doProcessOutputParam(OutputQueryParam outputSqlParam, Object paramValue) throws SQLException {
     if (paramValue instanceof ResultSet) {
       paramValue = resultSetHandler.processResultSet(connection, (ResultSet) paramValue);
-    } else if (paramValue instanceof SQLXML) {
-      SQLXML sqlxml = (SQLXML) paramValue;
-
-      paramValue = sqlxml.getString();
+    } else {
+      paramValue = processValue(paramValue);
     }
 
     return new OutputParamResult(outputSqlParam.getName(), paramValue);
   }
 
+  private Object processValue(Object paramValue) throws SQLException {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Refactor this method to reduce its Cognitive Complexity from 16 to the 15 allowed. ([squid:S3776](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS3776))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-core-extensions-mule-db-connector&pullRequest=160&issues=AWm_tKagotIXnmsUwvaO&open=AWm_tKagotIXnmsUwvaO)

[ISSUE_KEY]: AWm_tKagotIXnmsUwvaO